libioc/Provisioning/puppet.py

```diff
@@ -169,91 +171,122 @@ def provision(
     else:
         started = False
 
+    # Install packages
     pkg = libioc.Pkg.Pkg(
         logger=self.jail.logger,
         zfs=self.jail.zfs,
         host=self.jail.host
     )
-
+    yield from pkg.fetch(
```

Comment:
i didn't realise that install doesn't fetch, but given the architecture it makes sense